Travisci support

posborne opened the pull request:
These changes add support for running the project unit tests on Travis CI.  The build results can be viewed at https://travis-ci.org/digidotcom/python-wvalib and will also be linked in the badge (the badge shows status for the master branch which is currently not available).  In addition, build status should also be shown in PRs (we'll see shortly I suppose).

@tpmanley @mikewadsten @kellyschoenhofen  Please take a look.  If all looks good I'll merge.

Commit:
travisci: Add a basic travis configuration for running tests

Commit:
travisci: remove coverage from dev requirements to prevent travisci issues

Developers may still install coverage themselves but it will no
longer be included in the dev-requirements automatically.

Commit:
travisci: disable python3.4 tox env

There is an issue with certain versions of python 3.4 that results
in hanging unit tests.  This is not the case in the latest minor
revision of python 3.4 but is an issue with the python 3.4
included in many distributions (and running with the travis container).

Commit:
travisci: re-enable python 3.4 testing with non-broken httpretty

HTTPretty issue https://github.com/gabrielfalcao/HTTPretty/issues/221
results in a hang with certain versions of the Python interpreter
due to a bug in 3.4.2: http://bugs.python.org/issue20386.  Since
we have control over the HTTPretty version and not the interpreter
used by Travis, we will just use HTTPretty 0.8.6 for now.

Commit:
docs: add badges with build status and information from PyPI

The only major piece that is missing right now is coverage information
from coveralls, but that can be added in the future.

tpmanley commented:
I take it that Travis doesn't need coverage to be explicitly included, but does it hurt to leave it in? It would be nice to have the option of using it locally.

mikewadsten commented on README.md:
```diff
@@ -1,6 +1,11 @@
+[![Build Status](https://travis-ci.org/digidotcom/python-wvalib.svg?branch=master)](https://travis-ci.org/digidotcom/python-wvalib)
+[![Latest Version](https://pypip.in/version/wva/badge.svg)](https://pypi.python.org/pypi/wva/)
+[![Supported Python versions](https://pypip.in/py_versions/wva/badge.svg)](https://pypi.python.org/pypi/wva/)
+[![License](https://pypip.in/license/devicecloud/badge.svg)](https://pypi.python.org/pypi/devicecloud/)
```
I realize they're the same license (MPL 2.0) but should these links be `wva` and not `devicecloud`?

posborne replied:
Ah, good catch.  That's where I copied from but I forgot to replace one reference.  I'll fix that.

posborne commented:
@tpmanley, I was running into issues as coveralls has a dependency on coverage.  With both present, the TravisCI build would fail, so for now I am leaving it out.  Here's the failing build with both present: https://travis-ci.org/digidotcom/python-wvalib/jobs/56633023

Another solution might be to add back coverage to dev-requirements.txt frozen to a compatible version (e.g. 3.7.1) and remove it from tox.ini.  That might work well.

Commit:
readme: fix license badge to point to correct package (wva)

tpmanley commented:
Roger that, the changes looks good to me

mikewadsten commented:
:+1:

Commit:
travisci: add back coverage to dev-requirements and remove from tox.ini

The problem seen previously had to do with coverage being required
with different specs from different places.  We would still like
developer's to have coverage installed, so we remove the
requirement from tox and add it to dev-requirements.txt
